### PR TITLE
Safer way to handle unlock command of cli_wallet #1171

### PR DESCRIPTION
**Pull request by cogutvalera**
Check if exists regex expression, this PR is for https://github.com/bitshares/bitshares-core/issues/1171 issue

### CODE REVIEW

**Comment by cogutvalera** [comment]
after merging this https://github.com/bitshares/bitshares-fc/pull/82 PR and without merging PR https://github.com/bitshares/bitshares-core/pull/1382 there will be SEGFAULT when starting `cli_wallet`, this PR fixes it.

File: src/rpc/cli.cpp [hunk 1]
```diff
@@ -210,7 +210,7 @@ static int cli_completion(char *token, char ***array)
  */
 static int cli_check_secret(const char *source)
 {
-   if (boost::regex_match(source, cli_regex_secret()))
+   if (nullptr != cli_regex_secret().expression() && boost::regex_match(source, cli_regex_secret()))
```
Comment (pmconrad) [inline src/rpc/cli.cpp]: Please use https://www.boost.org/doc/libs/1_56_0/libs/regex/doc/html/boost_regex/ref/basic_regex.html#boost_regex.basic_regex.empty for checking.

Comment (cogutvalera) [reply src/rpc/cli.cpp]: Ok sure ! Thanks !

Comment (cogutvalera) [reply src/rpc/cli.cpp]: Done

**Review by pmconrad — APPROVED** [review]
Thanks!

**Pull request merged**